### PR TITLE
HTTP Range support

### LINKED ISSUE #5: HTTP RANGE support and copy link to clipboard
Hi,

Thanks for the very useful app.

I needed HTTP RANGE support in sharingpath, so I twiddled a bit with the code and came up 
with the diff below. Also, I implemented a naive way for copying the link to clipboard - but it works. Please have a look, maybe it'll be useful. I'm new to the github thing, so not sure how I should submit patches, sorry if it's messy.

All the best!

The diff:
```diff
diff -ur sharingpath.orig/js/script.js sharingpath/js/script.js
--- sharingpath.orig/js/script.js	Mon Oct 28 10:40:30 2019
+++ sharingpath/js/script.js	Tue Oct 29 09:44:02 2019
@@ -17,5 +17,38 @@
       window.open(path);
     }
   });
+  OCA.Files.fileActions.registerAction({
+    name: 'copy-sharing-path-link',
+    displayName: ['zh-CN', 'zh-HK', 'zh-TW', 'ja', 'ko'].includes(OC.getLanguage()) ? (t('files', 'Copy') + t('files_sharing', 'Sharing') + t('files', 'Path')) : (t('files', 'Copy') + ' ' + t('files_sharing', 'Sharing') + ' ' + t('files', 'Path')),
+    mime: 'file',
+    permissions: OC.PERMISSION_READ,
+    iconClass: 'icon-public',
+    actionHandler: function (filename, context) {
+      if (context.$file.data('shareTypes') !== OC.Share.SHARE_TYPE_LINK) {
+        OC.dialogs.info(t('files_sharing', 'No shared links'), t('gallery', 'Warning'));
+        return;
+      }
 
+      var path = OC.getProtocol() + '://' + OC.getHost() + OC.generateUrl('/apps/sharingpath/' + OC.getCurrentUser().uid + (context.dir === '/' ? '' : context.dir) + '/' + filename);
+
+      var dummyPath = document.createElement('textarea');
+      dummyPath.value = path;
+      dummyPath.setAttribute('readonly', '');
+      dummyPath.style.position = 'absolute';
+      dummyPath.style.left = '-9999px';
+      document.body.appendChild(dummyPath);
+      const selected =            
+          document.getSelection().rangeCount > 0        // Check if there is any content selected previously
+            ? document.getSelection().getRangeAt(0)     // Store selection if found
+            : false;
+
+      dummyPath.select();
+      document.execCommand("copy");
+      document.body.removeChild(dummyPath);
+      if (selected) {                                 // If a selection existed before copying
+          document.getSelection().removeAllRanges();    // Unselect everything on the HTML document
+          document.getSelection().addRange(selected);   // Restore the original selection
+      }
+    }
+  });
 });
diff -ur sharingpath.orig/lib/Controller/PathController.php sharingpath/lib/Controller/PathController.php
--- sharingpath.orig/lib/Controller/PathController.php	Mon Oct 28 10:40:30 2019
+++ sharingpath/lib/Controller/PathController.php	Tue Oct 29 09:45:13 2019
@@ -4,6 +4,7 @@
 
 use OC;
 use OC_Response;
+use OC_Files;
 use OC\Files\Filesystem;
 use OC\Share\Share;
 use OCP\Files\IRootFolder;
@@ -69,20 +70,18 @@
 
             $path = $userFolder->getRelativePath($file->getPath());
 
-            // output file contents without Content-Disposition header
-            header('Content-Transfer-Encoding: binary', true);
-            header('Pragma: public');
-            header('Expires: 0');
-            header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
-
             \OC_Util::setupFS($uid);
-            $fileSize = Filesystem::filesize($path);
-            $type     = OC::$server->getMimeTypeDetector()->getSecureMimeType(Filesystem::getMimeType($path));
-            if ($fileSize > -1) {
-                OC_Response::setContentLengthHeader($fileSize);
-            }
-            header('Content-Type: ' . $type, true);
-            Filesystem::getView()->readfile($path);
+            $dir = dirname($path);
+            $files_list = basename($path);
+            $server_params = array();
+            /**
+             * Http range requests support
+             */
+            if (isset($_SERVER['HTTP_RANGE'])) {
+		$server_params['range'] = \OC::$server->getRequest()->getHeader('Range');
+            } 
+
+            OC_Files::get($dir, array($files_list), $server_params);           
         } catch (NotFoundException $e) {
             http_response_code(404);
             exit;
```

### PR DESCRIPTION
**Pull request by wizardeur**
Updated patch for HTTP Range, without Content-Disposition header. Code mostly copied from https://github.com/nextcloud/server/blob/stable17/lib/private/legacy/files.php with minimal changes.

Fixes #5

### CODE REVIEW

File: lib/Controller/PathController.php [hunk 1]
```diff
@@ -4,89 +4,319 @@
...
+
+			// output file contents without Content-Disposition header
+			//            header('Content-Transfer-Encoding: binary', true);
+			//            header('Pragma: public');
+			//            header('Expires: 0');
+			//            header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+
+			\OC_Util::setupFS($uid);
+			$dir = dirname($path);
+			$files = basename($path);
```
Comment (rookie0) [inline lib/Controller/PathController.php]: no need pass $files

File: lib/Controller/PathController.php [hunk 2]
```diff
@@ -4,89 +4,319 @@
...
+	public static function get($dir, $files, $params = null) {
+		$view = \OC\Files\Filesystem::getView();
+		$getType = self::FILE;
+		try {
+			$filename = $dir . '/' . $files;
+			if (!$view->is_dir($filename)) {
+				self::getSingleFile($view, $dir, $files, is_null($params) ? array() : $params);
+				return;
+			}
+			set_time_limit($executionTime);
```
Comment (rookie0) [inline lib/Controller/PathController.php]: undefined var $executionTime

File: lib/Controller/PathController.php (hunk 1 above)
Comment (wizardeur) [reply lib/Controller/PathController.php]: Hi, still new to the github thingy ;) I just commited fixes to the pull request, please review. This point should be fixed (among other things).

File: lib/Controller/PathController.php (hunk 2 above)
Comment (wizardeur) [reply lib/Controller/PathController.php]: I just commited fixes to the pull request, please review. This point should be fixed as well.

**Pull request closed without merging**